src/gluonts/shell/sagemaker/dyn.py

```diff
@@ -78,10 +84,10 @@ def install_requirement(self, path: Path):
     def install(self, path):
         if path.is_file():
             if tarfile.is_tarfile(path):
-                self.handle_archive(tarfile.open, path)
+                self.handle_archive(tarfile.open, safe_extractall, path)
 
             elif zipfile.is_zipfile(path):
-                self.handle_archive(zipfile.ZipFile, path)
+                self.handle_archive(zipfile.ZipFile, extractall, path)
```

Comment:
```suggestion
                self.handle_archive(zipfile.ZipFile, zipfile.ZipFile.extractall, path)
```

This should also work.